test/util/lifecycle.js

```diff
@@ -15,15 +15,16 @@ import createTestContainer from './container';
 from the vm property of the Vue Test Utils wrapper: see
 https://github.com/vuejs/test-utils/issues/2140. In practice, the only global
 properties that we try to access in testing are $router and $route. We can
-provide access to them by shadowing the global properties with computed
-properties from a mixin. */
+provide access to them by copying $router to each component, then computing
+$route based on $router. */
 const shadowRouterProps = {
+  created() {
+    // eslint-disable-next-line no-self-assign
+    if (this.$router != null) this.$router = this.$router;
```

Comment:
I think this works because `$router` is a property on the prototype chain. By assigning it to `this`, we give `this` its own `$router` property. The `vm` property of the Vue Test Utils wrapper seems to provide access to an instance's own properties, but not global properties.